src/BackpackServiceProvider.php

```diff
@@ -99,11 +101,14 @@ public function register()
         $this->commands($this->commands);
     }
 
-    public function registerMiddlewareGroup(Router $router)
+    public function registerMiddlewareGroup(Router $router, \Illuminate\Contracts\Http\Kernel $kernel)
     {
         $middleware_key = config('backpack.base.middleware_key');
         $middleware_class = config('backpack.base.middleware_class');
 
+        // register the backpack error views
+        $kernel->pushMiddleware(BackpackErrorViews::class);
```

Comment:
Hmm from what I understand, we're force-pushing this middleware to all requests?

Wouldn't it be better if we include it in `config/backpack/base.php`, where we have:

```php
    // The classes for the middleware to check if the visitor is an admin
    // Can be a single class or an array of clases
    'middleware_class' => [
        App\Http\Middleware\CheckIfAdmin::class,
        \Illuminate\Foundation\Http\Middleware\ConvertEmptyStringsToNull::class,
        // \Backpack\CRUD\app\Http\Middleware\UseBackpackAuthGuardInsteadOfDefaultAuthGuard::class,
    ],
```

That would work the same way, or even better, right? Then the `BackpackErrorViews` would only run on admin routes (100%) and if someone doesn't like it, they can remove it.

Of course, this would mean another step in the upgrade guide, because people upgrading to v6 would need to manually add it to their config file. But I think that's ok, they'll need to delete the error views anyway.

Reply:
Plus, it would mean we no longer need to pass $kernel all the way to this method, right?

Reply:
There kernel is there after attempting to implement the solution you proposed and it wouldn't work as you said you expect in other comment:
> How about 404 errors? Those don't require the admin to be logged in right?

So I assumed you would be thinking to show 404 errors within /backpack/admin namespace using backpack errors.

With the current implementation, this are the errors displayed:
- http://domain.com/xxxx - Laravel error views 
- http://domain.com/admin/xxx - Backpack views either with template or without (non-logged, or early errors like 404 or 405)

With your approach the error pages would be:
- http://domain.com/xxxx - Laravel error views 
- http://domain.com/admin/xxx - Laravel error if it's an "early exception" (404 route not found, 405 method not allowed etc) or Backpack errors if any other kind of http error (abort() etc)

I think it makes more sense the way you suggested now, but that's not the idea I got from previous iterations.

I will do those changes since we are on the same page now. 

Thanks 🙏